api/tests/Unit/Service/Auth/AuthServiceTest.php

```diff
@@ -179,6 +193,39 @@ public function testisSecretValidForRole($clientSecret, $role, $expectedResult)
         $this->assertEquals($expectedResult, $this->authService->isSecretValidForRole($role, $request));
     }
 
+    /** @test */
+    public function jWTIsValid()
+    {
+        $this->JWTService->shouldReceive('verify')->with('not-a.real-jwt')->andReturn(true);
```

Comment:
Might bee worth adding variations for the common JWT highjack attacks here. Setting `alg=none `on an otherwise valid token and so on. There's a good summary of them over here: https://www.invicti.com/blog/web-security/json-web-token-jwt-attacks-vulnerabilities/ and here: https://infosecwriteups.com/attacks-on-json-web-token-jwt-278a49a1ad2e

Reply:
Yeah, fair point - I'd taken the approach that the package we're using to generate JWTs makes it impossible to use a none algorithm but that doesn't mean a JWT coming in from another source won't be malicious.

I think this is going to mean me changing the JWT package we're using out as its physically impossible to override `alg` but I don't think this is a bad thing as [another package](https://github.com/lcobucci/jwt) I was looking at has a much friendlier APIwhich should make the code a bit more flexible in the future and easier to write tests for.

Reply:
`lcobucci/jwt` is the same package we use over in Sirius, so good for consistency.

There's a particular balance with authentication mechanisms where I want to be sure I have tests that can verify common attack vectors can't be used against my app and if the abstraction doesn't allow that you're at the mercy of the 3rd party's tests.